lib/graphql/relay/connection/list.ex

```diff
@@ -19,25 +17,28 @@ defmodule GraphQL.Relay.Connection.List do
     slice_start = meta[:slice_start] || 0
     list_length = meta[:list_length] || length(records)
     slice_end = slice_start + length(records)
-    before_offset = get_offset_with_default(before, list_length)
-    after_offset = get_offset_with_default(a_after, -1)
+    before_offset = cursor_to_offset(before, list_length)
+    after_offset = cursor_to_offset(a_after, -1)
     start_offset = Enum.max([slice_start - 1, after_offset, -1]) + 1
     end_offset = Enum.min([slice_end, before_offset, list_length])
 
-    if first do
-      end_offset = Enum.min([end_offset, start_offset + first])
+    end_offset = if first do
```

Comment:
Was wondering whether it was possible to refactor this to be set in a single `if`. Thoughts?